python/paddle/trainer_config_helpers/networks.py

```diff
@@ -1396,6 +1398,85 @@ def simple_attention(encoded_sequence,
         input=scaled, pooling_type=SumPooling(), name="%s_pooling" % name)
 
 
+@wrap_name_default()
+def dot_product_attention(encoded_sequence,
+                          attending_sequence,
+                          transformed_state,
+                          softmax_param_attr=None,
+                          name=None):
+    """
+    Calculate and return a context vector with dot-product attention mechanism.
+    Size of the context vector equals to size of the attending_sequence.
+
+    ..  math::
+
+        a(s_{i-1},h_{j}) & = s_{i-1}^\mathrm{T} h_{j}
+
+        e_{i,j} & = a(s_{i-1}, h_{j})
+
+        a_{i,j} & = \\frac{exp(e_{i,j})}{\\sum_{k=1}^{T_x}{exp(e_{i,k})}}
+
+        c_{i} & = \\sum_{j=1}^{T_{x}}a_{i,j}z_{j}
+
+    where :math:`h_{j}` is the jth element of encoded_sequence,
+    :math:`z_{j}` is the jth element of attending_sequence,
+    :math:`s_{i-1}` is transformed_state
+
+    The example usage is:
+
+    ..  code-block:: python
+
+        context = dot_product_attention(encoded_sequence=enc_seq,
+                                        attending_sequence=att_seq,
+                                        transformed_state=state,)
+
+    :param name: name of the dot-product attention model.
+    :type name: basestring
+    :param softmax_param_attr: parameter attribute of sequence softmax
+                               that is used to produce attention weight.
+    :type softmax_param_attr: ParameterAttribute
+    :param encoded_sequence: output of the encoder
+    :type encoded_sequence: LayerOutput
+    :param attending_sequence: attention weight is computed by a feed forward neural
+                               network which has two inputs : decoder's transformed
+                               hidden state of previous time step and encoder's output.
+                               attending_sequence is the sequence to be attended.
+    :type attending_sequence: LayerOutput
+    :param transformed_state: transformed hidden state of decoder in previous time step,
+                              its size should equal to encoded_sequence's. Here we do the
```

Comment:
whose dimension should be equal to encoded_sequence's dimension.  Or use a period at the end of last sentence and changes "its" into "Its".

Reply:
Done